core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -739,6 +813,61 @@ public GroupByIdBlock getResult()
         }
     }
 
+    private class GetLowCardinalityDictionaryGroupIdsWork
+            extends LowCardinalityDictionaryWork<GroupByIdBlock>
+    {
+        private final long[] blockData;
+        private boolean finished;
+
+        public GetLowCardinalityDictionaryGroupIdsWork(Page page)
+        {
+            super(page);
+            blockData = new long[page.getPositionCount()];
+        }
+
+        @Override
+        public boolean process()
+        {
+            int positionCount = page.getPositionCount();
+            checkState(lastPosition < positionCount, "position count out of bound");
+
+            int[] combinations = new int[cardinality];
```

Comment:
this does not store combinations but rather a sample position for given combination